networkit/cpp/distance/test/FloydWarshallGTest.cpp

```diff
@@ -0,0 +1,415 @@
+    }
+
+    Graph directedGraphWithNegativeSelfLoop() {
+        Graph graph(5, true, true);
+        graph.addEdge(0, 1, 3);
+        graph.addEdge(1, 1, -2); // self-loop with negative cycle
+        graph.addEdge(1, 2, 2);
+        graph.addEdge(2, 3, 1);
+        graph.addEdge(3, 4, 4);
+        graph.addEdge(4, 0, 1);
+        return graph;
+    }
+
+    Graph oddNodeEdgesHaveZeroWeightGraph() {
+        Graph graph(100, true);
+        for (node u = 0; u < 98; ++u) {
+            // All weights between odd edge IDs are zero
+            // All other weights are one
+            if (u & 1) {
+                graph.addEdge(u, u + 1, 1);
+                graph.addEdge(u, u + 2, 0);
+            } else {
+                graph.addEdge(u, u + 1, 1);
+                graph.addEdge(u, u + 2, 1);
+            }
+        }
+        graph.addEdge(98, 99, 1);
+        return graph;
+    }
+};
+
+TEST_F(FloydWarshallGTest, testConstructorThrowsUnweightedGraph) {
+    Graph graph(1, false);
+    try {
+        FloydWarshall test(graph);
+        FAIL() << "Expected std::runtime_error";
+    } catch (const std::runtime_error &e) {
+        EXPECT_STREQ(e.what(), "The input graph is unweighted!");
+    } catch (...) {
+        FAIL() << "Expected std::runtime_error but got a different exception.";
+    }
+}
+
+TEST_F(FloydWarshallGTest, testGetDistanceThrows) {
+    Graph graph(1, true);
+    FloydWarshall test(graph);
+    try {
+        test.getDistance(0, 1);
+        FAIL() << "Expected std::runtime_error";
+    } catch (const std::runtime_error &e) {
+        EXPECT_STREQ(e.what(), "Error, run must be called first");
+    } catch (...) {
+        FAIL() << "Expected std::runtime_error but got a different exception.";
+    }
+}
+
+TEST_F(FloydWarshallGTest, testIsNodeInNegativeCycleThrows) {
+    Graph graph(1, true);
+    FloydWarshall test(graph);
+    try {
+        test.isNodeInNegativeCycle(0);
+        FAIL() << "Expected std::runtime_error";
+    } catch (const std::runtime_error &e) {
+        EXPECT_STREQ(e.what(), "Error, run must be called first");
+    } catch (...) {
+        FAIL() << "Expected std::runtime_error but got a different exception.";
+    }
+}
+
+TEST_F(FloydWarshallGTest, testGetNodesOnShortestPathThrows) {
+    Graph graph(2, true);
+    FloydWarshall test(graph);
+    try {
+        test.getNodesOnShortestPath(0, 1);
+        FAIL() << "Expected std::runtime_error";
+    } catch (const std::runtime_error &e) {
+        EXPECT_STREQ(e.what(), "Error, run must be called first");
+    } catch (...) {
+        FAIL() << "Expected std::runtime_error but got a different exception.";
+    }
+}
+
+TEST_F(FloydWarshallGTest, testGetDistanceCompleteGraphK3) {
+    auto graph = completeGraphK3();
+    FloydWarshall test(graph);
+    test.run();
+    const std::vector<std::vector<edgeweight>> expectedDistances{{0, 1, 3}, {1, 0, 2}, {3, 2, 0}};
+    for (node source = 0; source < graph.numberOfNodes(); ++source) {
```

Comment:
Since this double loop is used in multiple tests, implement a helper function to check that the distances are as expected.

Reply:
Even better: implement a helper function that takes as input a graph and the expected distances, and tests that they are as expected.